### PR TITLE
SQL-1674: Remove deprecated format, formatVersion, and dialect options from ODBC driver

### PR DESCRIPTION
**Pull request by EthanHardyMongo**

### CODE REVIEW

File: shared_sql_utils/src/dsn.rs [hunk 1]
```diff
@@ -228,14 +228,9 @@ impl Dsn {
...
-            .map(|(key, value)| {
-                if value.is_empty() {
-                    return "".into();
-                }
-                format!("{key}={value};")
-            })
-            .collect::<String>()
+        self.iter().fold("".into(), |acc, (key, value)| {
+            format!("{acc}{key}={value};")
+        })
```
Comment (EthanHardyMongo) [inline shared_sql_utils/src/dsn.rs:233]: Clippy made me do this.

**Review by bucaojit — COMMENTED** [review]
@EthanHardyMongo changes look good and nice to see the integration test pass.  Let's hold off on merging this just yet until I can confirm that the change has made it in time for the ADF release today.  Otherwise it would have to wait until the next Tuesday release.

**Comment by bucaojit** [comment]
As mentioned on the JDBC PR:
The MHOUSE-8151 change didn't make it into the mongohouse [branch v20231107](https://github.com/10gen/mongohouse/tree/v20231107).  So I added the tag for fix version [v20231114](https://jira.mongodb.org/issues/?jql=project+%3D+MHOUSE+AND+fixVersion+%3D+v20231114) which is next Tuesday.  Let's hold off on merging this until after that is released.

**Review by terakilobyte — APPROVED** [review]
LGTM.

File: shared_sql_utils/src/dsn.rs (hunk 1 above)
Comment (EthanHardyMongo) [reply shared_sql_utils/src/dsn.rs]: Matt fixed this in a different PR while this was blocked.

**Comment by EthanHardyMongo** [comment]
evergreen merge

**Pull request merged**